Piped crashes when multiple Kubernetes providers are pointing to the same actual cluster

**What happened**:

```
panic: close of closed channel

goroutine 45 [running]:
github.com/pipe-cd/pipe/pkg/app/piped/livestatestore/kubernetes.(*Store).Run(0xc000684230, 0x26b8750, 0xc0009d7ac0, 0xa, 0xc0005aa088)
	pkg/app/piped/livestatestore/kubernetes/kubernetes.go:117 +0x5fa
github.com/pipe-cd/pipe/pkg/app/piped/livestatestore.(*store).Run.func1(0x0, 0x0)
	pkg/app/piped/livestatestore/livestatestore.go:133 +0x87
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc000758ff0, 0xc00057a2e0)
	external/org_golang_x_sync/errgroup/errgroup.go:57 +0x59
created by golang.org/x/sync/errgroup.(*Group).Go
	external/org_golang_x_sync/errgroup/errgroup.go:54 +0x66
```

**What you expected to happen**:

**How to reproduce it**:

**Environment**:
- `piped` version:
- `control-plane` version:
- Others:

--- a/pkg/app/piped/livestatestore/livestatestore.go
+++ b/pkg/app/piped/livestatestore/livestatestore.go
@@ -134,32 +134,37 @@ func (s *store) Run(ctx context.Context) error {
 	group, ctx := errgroup.WithContext(ctx)
 
 	for i := range s.kubernetesStores {
+		cpName := i
 		group.Go(func() error {
-			return s.kubernetesStores[i].Run(ctx)
+			return s.kubernetesStores[cpName].Run(ctx)
 		})
 	}
 
 	for i := range s.terraformStores {
+		cpName := i
 		group.Go(func() error {
-			return s.terraformStores[i].Run(ctx)
+			return s.terraformStores[cpName].Run(ctx)
 		})
 	}
 
 	for i := range s.cloudrunStores {
+		cpName := i
 		group.Go(func() error {
-			return s.cloudrunStores[i].Run(ctx)
+			return s.cloudrunStores[cpName].Run(ctx)
 		})
 	}
 
 	for i := range s.lambdaStores {
+		cpName := i
 		group.Go(func() error {
-			return s.lambdaStores[i].Run(ctx)
+			return s.lambdaStores[cpName].Run(ctx)
 		})
 	}
 
 	for i := range s.ecsStores {
+		cpName := i
 		group.Go(func() error {
-			return s.ecsStores[i].Run(ctx)
+			return s.ecsStores[cpName].Run(ctx)
 		})
 	}
 

